File: lib/core_services/eiga/interfaces/eiga_context_with_episodes.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import '../export.dart';

part 'eiga_context_with_episodes.freezed.dart';
part 'eiga_context_with_episodes.g.dart';

@freezed
sealed class EigaContextWithEpisodes with _$EigaContextWithEpisodes {
  const factory EigaContextWithEpisodes({
    required String eigaId,
    required MetaEiga metaEiga,
    Season? season,
    required List<EigaEpisode> episodes,
  }) = _EigaContextWithEpisodes;

  factory EigaContextWithEpisodes.fromJson(Map<String, dynamic> json) =>
      _$EigaContextWithEpisodesFromJson(json);
}

File: lib/core_services/eiga/interfaces/eiga_context_with_episodes.ts
import type { MetaEiga } from "./meta_eiga"
import type { Season } from "./meta_eiga"
import type { EigaEpisode } from "./eiga_episode"

export interface EigaContextWithEpisodes {
  eigaId: string
  metaEiga: MetaEiga
  season?: Season
  episodes: EigaEpisode[]
}
